src/ui/public/registry/field_formats.js

```diff
@@ -1,97 +1,106 @@
 import _ from 'lodash';
-import { uiRegistry } from 'ui/registry/_registry';
+import chrome from 'ui/chrome';
 import { FieldFormat } from '../../field_formats/field_format';
+import { IndexedArray } from 'ui/indexed_array';
 
-export const RegistryFieldFormatsProvider = uiRegistry({
-  name: 'fieldFormats',
-  index: ['id'],
-  group: ['fieldType'],
+class FieldFormatRegistry extends IndexedArray {
+  constructor(config) {
+    const spec = config && config.spec;
+    const iaOpts = _.defaults(_.pick(spec, IndexedArray.OPT_NAMES), { index: ['id'] });
+    super(iaOpts);
 
-  invokeProviders(providers) {
-    // in order to ensure that FieldFormats can be instantiated on the
-    // server and the browser we don't provide them access to the Angular
-    // injector, just the FieldFormat class.
-    return providers.map(createSomeFormat => (
-      createSomeFormat(FieldFormat)
-    ));
-  },
+    this._config = chrome.getUiSettingsClient();
+    this.getConfig = (...args) => this._config.get(...args);
+    this._defaultMap = [];
+    this._providers = [];
+    this.init();
+  }
 
-  constructor: function (config) {
-    const getConfig = (...args) => config.get(...args);
-    const self = this;
-    let defaultMap;
+  init() {
+    this.parseDefaultTypeMap(this._config.get('format:defaultTypeMap'));
 
-    function init() {
-      config.watch('format:defaultTypeMap', parseDefaultTypeMap);
-    }
+    this._config.subscribe(({ key, newValue }) => {
```

Comment:
@spalger: i am not sure why the watch was used here and i couldn't get this subscribe to fire  ... do we need this ? if so does it look correct ? when would we need to unsubscribe ?

Reply:
It is called whenever the config changes, seems to be working when I test it:

![image](https://user-images.githubusercontent.com/1329312/38392966-20ff9878-38de-11e8-9896-44c2d2ba9b3c.png)

> when would we need to unsubscribe ?

That's a good question... I don't think we need to unsubscribe since the fieldFormatsRegistry is never torn down, but I'm concerned how that will work with tests, especially since the `uiSettingsClient` is currently stubbed out for each test https://github.com/elastic/kibana/blob/master/src/ui/public/test_harness/test_harness.js#L33-L49

Reply:
How about we add this as another method on the `chrome` object, like https://github.com/elastic/kibana/blob/master/src/ui/public/chrome/api/ui_settings.js#L12-L14, and then stub it out in the tests just like we do for the `uiSettings` client. Then I'll prioritize migrating the `uiSettings` and `fieldFormats` services over to the new platform once we have that plumbing in place.